spot_driver/include/spot_driver/api/default_spot_api.hpp

```diff
@@ -22,15 +23,15 @@ class DefaultSpotApi : public SpotApi {
   tl::expected<void, std::string> createRobot(const std::string& ip_address, const std::string& robot_name) override;
   tl::expected<void, std::string> authenticate(const std::string& username, const std::string& password) override;
   tl::expected<bool, std::string> hasArm() const override;
-  std::shared_ptr<KinematicApi> kinematicApi() const override;
+  std::optional<std::shared_ptr<KinematicApi>> kinematicInterface() const override;
```

Comment:
Are you using the `nullopt` in a different capacity that `nullptr`?

Reply:
I'm sort of on the fence about this still. What I landed on before pushing was that a `nullopt` within a `std::optional` is communicating something different than a `nullptr` within a `std::shared_ptr`, but now I'm leaning towards both of those meaning "this hasn't been initialized yet". I think I'll change it back and just check for `nullptr` when initializing the IK node, since that's a simpler change and a pattern that can be used for the other nodes too.